app/models/usage_threshold.rb

```diff
@@ -7,16 +7,18 @@ class UsageThreshold < ApplicationRecord
   self.discard_column = :deleted_at
 
   belongs_to :organization
-  belongs_to :plan
+  belongs_to :plan, optional: true
+  belongs_to :subscription, optional: true
 
   has_many :applied_usage_thresholds
   has_many :invoices, through: :applied_usage_thresholds
 
-  monetize :amount_cents, with_currency: ->(threshold) { threshold.plan.amount_currency }
+  monetize :amount_cents, with_currency: ->(threshold) { threshold.currency }
 
   validates :amount_cents, numericality: {greater_than: 0}
-  validates :amount_cents, uniqueness: {scope: %i[plan_id recurring deleted_at]}, if: -> { deleted_at.nil? }
-  validates :recurring, uniqueness: {scope: %i[plan_id deleted_at]}, if: -> { recurring? && deleted_at.nil? }
+  validates :amount_cents, uniqueness: {scope: %i[plan_id recurring deleted_at]}, if: -> { deleted_at.nil? && subscription_id.nil? }
+  validates :recurring, uniqueness: {scope: %i[plan_id deleted_at]}, if: -> { recurring? && deleted_at.nil? && subscription_id.nil? }
```

Comment:
I intend to get rid of these uniqueness rails validation (which triggers query) when I rework the service.